features/FxA-16-devices-view/README.md

```diff
@@ -4,5 +4,174 @@ Device View - Basic UI
+may have been used without my permission,
+to check what devices are connected
+and to disconnect devices that I do not recognize.
+
+
+UI Mocks
+--------
+
+@ryanfeeley's mocks go here
+
+
+Details
+-------
+
+#### What is a "device"?
+
+By "device" here we mean "an instance of Firefox".
+There is no separate notion of having multiple "browsers" on a single "device".
+Technically the concept is closer to "profiles"
+but this is not a good name to present the users.
+
+* When I use a desktop machine
+  and connect both Firefox Release and Firefox Nightly to my account,
+  they appear as two separate devices.
+  There will be no indicatiion that they're on the same physical machine
+  apart from similarity in their auto-generated name and their UA details.
+
+* When I use Firefox for Android or Firefox for iOS...
+  TODO what exactly do we expect to happen?
+
+* When I use a FirefoxOS device,
+  it appears in the view as a single device
+  since the connetion is managed at the system level.
+  Also, TODO do we care at this stage?
+
+For the initial version, only things that connect to sync count as devices.
+
+* When I log in to Marketplace with my Firefox Account,
+  using the web-based flow in a browser that is not connected to my account,
+  the resulting session does *not* appear in the list of devices.
+
+* When I log in to Pocket with my Firefox Account
+  using non-Firefox web browser,
+  the resulting session does *not* appear in the list of devices.
+
+* When I have existing Firefox instances
+  that are already connected to sync,
+  they will appear in the devices view automatically.
+
+* TODO this also means that FirefoxOS should not be in the list
+  because it doesn't connect to sync yet.
+  It will soon though and it's something for us to think about.
+  
+
+
+#### What details do we show about the device?
+
+For each device we will show:
+
+* The human-readable name for the device,
+  either an auto-generated one or one provided by the user.
+
+* The time it was last seen,
+  i.e. the times at which it last used its session token
+  to interact with the FxA servers.
+
+* The browser type and version,
+  and operating system platform and version,
+  where this information is available in the user-agent string
+  provided by the device.
+
+* Whether or not the device's login state is still valid
+  (it could be  in the "need to re-authenticate" state after password reset).
+
+* Whether it is the device currently viewing the devices view.
+
+
+#### Disconnecting devices
+
+When I sign out of Firefox Sync through the browser's UI,
+the session token belonging to that device is destroyed
+and all information about the device is removed from the server.
+It no longer appears in the devices view.
+
+When I view the devices list from a device that is connected to my account,
+I do not have the ability to disconnect that device,
+because this would be a confusing user experience.
+I must disconnect it through the device's native UI.
```

Comment:
I was thinking about this last night too. We could in theory allow the current device to disconnect itself, and why not, we are allowing the user to sign out of a remote device. I theorize its just as natural to sign out of the current device as force a signout of a remote device. We'd have to [notify the browser](https://github.com/mozilla/fxa-content-server/issues/2993), and maybe it makes sense to re-add the signout link for Sync.